External/SPMmodified/xASL/xASL_io_ReadNifti.m

```diff
@@ -88,10 +95,14 @@
 
 pathOut = [pathIn ImExt]; % put it back
     
-%% Load JSON sidecar if it exists
-
-if exist(pathJson)
-    json = xASL_io_ReadJson(pathJson);
+%% Load JSON sidecar if it exists and if the output is required
+if nargout > 2
```

Comment:
Yes - it doesn't need to be there, I agree. But it's to speed up things, limit disc access. Makes no huge difference in the end of course, but it's a good practice to avoid doing unnecessary things.

So given that we have clear comments in the code about this, I'd say that it doesn't make the code more complicated. 

So I'd indeed keep this as I feel strong about this. We have improved our code a lot to be readable, but we have a lot to work on on optimal processing speed etc. This is the first small step :) You can resolve this - I wanted you to hear this explanation...

Reply:
Of course I agree.

We aim for a good balance between speed and complexity of course.

So tiny speed-up tricks are only used in function called a lot where tiny improvements and high optimization matters. In bigger functions with large overhead, we don't bother with such things of course.

So I fully agree with you, it would not even occur to me to do it differently :)